image_tool.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+from matplotlib.mlab import bivariate_normal
+
+from bokeh.core.properties import Instance
+
+from bokeh.models import ColumnDataSource, ColorBar, LinearColorMapper, BoxAnnotation, ColorMapper
+from bokeh.models import Range1d, LinearColorMapper, LogColorMapper, LogTicker
+from bokeh.models.tools import Drag, DEFAULT_BOX_OVERLAY
+from bokeh.plotting import figure, show
+from bokeh.palettes import Viridis256
+
+### custom tool definition
+
+DEFAULT_COLOR_BAR = lambda: ColorBar(location=(0,0), color_mapper=LinearColorMapper(palette='Spectral9', low=0, high=10))
+
+class ColorBarTool(Drag):
+    __implementation__ = open('color_bar_tool.coffee').read()
```

Comment:
Since 0.12.3 you can use simply `__implementation__ = 'color_bar_tool.coffee'`. Oneliner ending in `.coffee` is interpreted as file path.